services/source-debugger/eventUploader.go

```diff
@@ -137,17 +126,20 @@ func uploadEvents(eventBuffer []*GatewayEventBatchT) {
 
 			//updating originalTimestamp in the event using the formula
 			//receivedAt - (sentAt - originalTimeStamp)
-			orgTS, err := time.Parse(time.RFC3339, ev.OriginalTimestamp)
+			originalTimestamp := ev["originalTimestamp"].(string)
+			sentAt := ev["sentAt"].(string)
```

Comment:
same err handling as above

Reply:
also maybe add sentAt in gateway if not present in payload
something like we do for receivedAt https://github.com/rudderlabs/rudder-server/blob/master/gateway/gateway.go#L395 but with additional if present check

Reply:
done. reusing the same logic from processor.go. As discussed, did not add it in gateway for now.